test/e2e/ratelimit/ratelimit_universal.go

```diff
@@ -0,0 +1,136 @@
+package ratelimit
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/config/core"
+	. "github.com/kumahq/kuma/test/framework"
+)
+
+func RateLimitOnUniversal() {
+	var cluster Cluster
+	var deployOptsFuncs []DeployOptionsFunc
+	rateLimitPolicy := `
+type: RateLimit
+mesh: default
+name: rate-limit-all-sources
+sources:
+- match:
+   kuma.io/service: "*"
+destinations:
+- match:
+   kuma.io/service: echo-server_kuma-test_svc_8080
+conf:
+ http:
+   connections: 5
+   interval: 10s
+   onError:
+     status: 423
+     headers:
+       - key: "x-kuma-rate-limited"
+         value: "true"
+`
+
+	BeforeEach(func() {
+		clusters, err := NewUniversalClusters(
+			[]string{Kuma3},
+			Silent)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Global
+		cluster = clusters.GetCluster(Kuma3)
+		deployOptsFuncs = KumaUniversalDeployOpts
+
+		err = NewClusterSetup().
+			Install(Kuma(core.Standalone, deployOptsFuncs...)).
+			Setup(cluster)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = cluster.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		demoClientToken, err := cluster.GetKuma().GenerateDpToken("default", "demo-client")
+		Expect(err).ToNot(HaveOccurred())
+
+		echoServerToken, err := cluster.GetKuma().GenerateDpToken("default", "echo-server_kuma-test_svc_8080")
+		Expect(err).ToNot(HaveOccurred())
+
+		err = YamlUniversal(rateLimitPolicy)(cluster)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = NewClusterSetup().
+			Install(EchoServerUniversal(AppModeEchoServer, "default", "universal", echoServerToken, WithTransparentProxy(true))).
+			Install(DemoClientUniversal(AppModeDemoClient, "default", demoClientToken, WithTransparentProxy(true))).
+			Setup(cluster)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if ShouldSkipCleanup() {
+			return
+		}
+		err := cluster.DeleteKuma(deployOptsFuncs...)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = cluster.DismissCluster()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should limit to 5 requests per 10 sec", func() {
+		time.Sleep(5 * time.Second)
```

Comment:
why sleep? This will be flaky IMHO. How about wrapping this in Eventually with interval of 11s